system/ui/widgets/network.py

```diff
@@ -150,21 +150,24 @@ def _draw_network_item(self, rect, network: NetworkInfo, clicked: bool):
     self._draw_signal_strength_icon(signal_icon_rect, network)
 
     if isinstance(self.state, StateIdle) and rl.check_collision_point_rec(rl.get_mouse_position(), ssid_rect) and clicked:
-      if not network.is_saved:
+      if not network.is_saved and network.security_type != SecurityType.OPEN:
         self.state = StateNeedsAuth(network)
       elif not network.is_connected:
         self.connect_to_network(network)
 
   def _draw_status_icon(self, rect, network: NetworkInfo):
     """Draw the status icon based on network's connection state"""
-    icon_file = ""
+    icon_file = None
     if network.is_connected:
       icon_file = "icons/checkmark.png"
     elif network.security_type == SecurityType.UNSUPPORTED:
```

Comment:
I think we should also add `SecurityType.WPA3` here since it's unsupported by us

Reply:
oh, we have a method for determining the SecurityType, I guess it can be fixed there instead.